CMakeLists.txt

```diff
@@ -33,55 +33,87 @@
 
 cmake_minimum_required (VERSION 3.0)
 
-project ("rttr" LANGUAGES CXX)
+project (rttr LANGUAGES CXX)
 
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMake")
 
 if (CMAKE_BUILD_TYPE STREQUAL "")
   # CMake defaults to leaving CMAKE_BUILD_TYPE empty. This screws up
   # differentiation between debug and release builds.
-  set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Choose the type of build, options are: None (CMAKE_CXX_FLAGS or CMAKE_C_FLAGS used) Debug Release RelWithDebInfo MinSizeRel." FORCE)
+  set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Choose the type of build" FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif ()
 
 # our little cmake helper functions
 include(utility)
 
 # set up option variable for cmake
-option(BUILD_RTTR_DYNAMIC "Build the dynamic/shared version of RTTR library" ON)
-option(BUILD_UNIT_TESTS "Build the unit tests of RTTR" ON)
-option(BUILD_STATIC "Build RTTR as static library" OFF)
-option(BUILD_WITH_STATIC_RUNTIME_LIBS "Link against the static runtime libraries" OFF)
-option(BUILD_WITH_RTTI "Enable build with C++ runtime type information for compilation" ON)
-option(BUILD_BENCHMARKS "Enable this to build the benchmarks" OFF)
-option(BUILD_EXAMPLES "Enable this to build the examples" On)
-option(BUILD_DOCUMENTATION "Enable this to build the documentation" ON)
-option(BUILD_INSTALLER "Enable this to build the installer" ON)
-option(BUILD_PACKAGE "Enable this to build the installer" ON)
-option(USE_PCH "Use precompiled header files for compilation" ON)
-option(CUSTOM_DOXYGEN_STYLE "Enable this option to use a custom doxygen style for HTML documentation; Otherwise the default will be used" ON)
-option(BUILD_WEBSITE_DOCU "Enable this option to create the special docu for the website" OFF)
+
+if (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
```

Comment:
For what is this `if` and the code below?

Reply:
This `if` will get fired only if you open this project `CMakeLists.txt` as a top, standalone project, but it will not if it's parsed for example from `add_subdirectory(vendor/rttr)`